src/business_logic/execution/objects.rs

```diff
@@ -12,6 +13,12 @@ pub(crate) struct OrderedEvent {
 
 pub(crate) struct TransactionExecutionContext {
     pub(crate) n_emitted_events: u32,
+    pub(crate) version: BigInt,
```

Comment:
version could be typed as `usize`, or there is a reason for which it should be BigInt? we try to avoid unnecessary use of of BigInt, for performance reasons.